addons/account/models/account_reconcile_model.py

```diff
@@ -501,7 +501,7 @@ def _get_invoice_matching_query(self, st_lines, excluded_ids=None, partner_map=N
                 (
                     move.invoice_payment_ref IS NOT NULL
                     AND
-                    TRIM(move.invoice_payment_ref) = TRIM(st_line.name)
+                    regexp_replace(move.invoice_payment_ref, '\W+', '', 'g') = regexp_replace(st_line.name, '\W+', '', 'g')
```

Comment:
You commit message is about removing whitespaces right? Why not `\s+` then?

Reply:
true